### PR TITLE
ES|QL: Set default query LIMIT to 1000

### LINKED ISSUE #105601: ESQL: increase default implicit limit to 1000
### Description

Change the current ESQL implicit limit from 500 to 1000, essentially doubling it.

### PR DESCRIPTION
**Pull request by luigidellaquila**
When an ES|QL query does not define a LIMIT explicitly, an implicit LIMIT is added by the planner.
This PR increases this default from 500 to 1000.

Resolves: https://github.com/elastic/elasticsearch/issues/105601

### CODE REVIEW

**Review by leemthompo — COMMENTED** [review]
Docs changes LGTM 👍

File: x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/EsqlTestUtils.java [hunk 1]
```diff
@@ -185,7 +185,8 @@ public static List<List<Object>> getValuesList(Iterator<Iterator<Object>> values
 
     public static List<String> withDefaultLimitWarning(List<String> warnings) {
         List<String> result = warnings == null ? new ArrayList<>() : new ArrayList<>(warnings);
-        result.add("No limit defined, adding default limit of [500]");
+        result.add("No limit defined, adding default limit of [1000]");
+        result.add("No limit defined, adding default limit of [500]"); // this is for bwc tests, the limit in v 8.12.x is 500
```
Comment (astefan) [inline x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/EsqlTestUtils.java:189]: Same here about specific versions check.

**Review by astefan — COMMENTED** [review]
Looks ok, but assertions we make in some tests related to bwc need to be more specific.

File: x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java [hunk 2]
```diff
@@ -817,7 +817,10 @@ private static HttpEntity assertWarnings(Response response, List<String> allowed
     }
 
     private static Set<String> mutedWarnings() {
-        return Set.of("No limit defined, adding default limit of [500]");
+        return Set.of(
+            "No limit defined, adding default limit of [1000]",
+            "No limit defined, adding default limit of [500]" // this is for bwc tests, the limit in v 8.12.x is 500
```
[the remark this answers is not in the document]
Comment (luigidellaquila) [reply x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java:822]: I'm on it, but it's not trivial apparently.
The problem is that in a mixed cluster scenario the warning can be one or the other based on the coordinator version. Most of our YAML tests already use `No limit defined, adding default limit of \\[.*\\]`, but I'm checking if there is a better way to do it.

File: x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/EsqlTestUtils.java (hunk 1 above)
Comment (luigidellaquila) [reply x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/EsqlTestUtils.java]: This specific check wasn't needed, it's never used by bwc tests

File: x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (luigidellaquila) [reply x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java:822]: After some code checks and debugging, I came to the conclusion that when we check the warnings we don't have any information about the version of the coordinator; the `response` doesn't contain that information, and the client we use for the tests uses `NodeSelector.ANY`, so in a mixed cluster scenario we can hit any node.

In conclusion, a possible solution would require a significant refactoring of the multi-node test infrastructure, and I'm not sure this single case is worth the effort.

**Review by costin — APPROVED** [review]
Thanks for updating all the files - Luigi!
Pending the BWC tests, LGTM.

**Review by astefan — APPROVED** [review]
LGTM

**Pull request merged**